### PR TITLE
Add template files

### PR DESCRIPTION
**Pull request by nikhita**
Ref: https://github.com/kubernetes/steering/issues/28 (for CONTRIBUTING.md) and https://github.com/kubernetes/community/issues/3053 (for CoC) and https://github.com/kubernetes/steering/issues/57#issuecomment-450132738 (for LICENSE)

The content is as per https://github.com/kubernetes/kubernetes-template-project

/assign @andrewsykim @cheftako @mcrute

### CODE REVIEW

**Comment by mcrute** [comment]
Thanks!

/lgtm
/approve

File: README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# cloud-provider-sample
+
+This is a sample of how to build a cloud provider repo. This will build a Kubernetes image which deploys on bare metal.
```
Comment (cheftako) [inline README.md]: This is the intent. I think it would be good to warn people its a WIP until its actually working.

Comment (nikhita) [reply README.md]: Oh good point! I have added a note. :+1:

**Comment by cheftako** [comment]
Thanks @nikhita

**Pull request merged**